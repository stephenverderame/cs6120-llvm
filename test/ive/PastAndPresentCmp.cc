// NOLINTBEGIN
#include <cstdio>
int main()
{
    int i = 0;
    while (i < 100) {
        int k = i * 10 + 5;
        if (i < 50 && k > 100) {
            printf("%d\n", k);
        }
        ++i;
        int m = i * 20 + 5;
        if (i >= 75 || i < 10) {
            printf("%d\n", m);
        }
    }
}
// NOLINTEND