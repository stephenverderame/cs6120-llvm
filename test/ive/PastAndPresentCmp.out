Loop: 
1:                                                ; preds = %20, %0
  %.0 = phi i32 [ 0, %0 ], [ %12, %20 ]
  %2 = icmp slt i32 %.0, 100
  br i1 %2, label %3, label %21

Basic IV:   %12 = add nsw i32 %.0, 1
Derived IV: %13 = %12 * 20
Derived IV: %14 = %12 * 20 + 5
Derived IV: %4 = .0 * 10
Derived IV: %5 = .0 * 10 + 5
Basic IV:   %.0 = phi i32 [ 0, %0 ], [ %12, %20 ]
